tee-worker/omni-executor/parentchain/listener/src/lib.rs

```diff
@@ -130,3 +147,60 @@ pub async fn create_listener<EthereumIntentExecutorT: IntentExecutor + Send + Sy
 		last_processed_log_repository,
 	)
 }
+
+#[allow(unused_assignments, unused_mut, unused_variables, clippy::type_complexity)]
+async fn perform_attestation(
+	client_factory: Arc<SubxtClientFactory<CustomConfig>>,
+	signer: Keypair,
+	transaction_signer: &Arc<
+		TransactionSigner<
+			SubstrateKeyStore,
+			SubxtClient<CustomConfig>,
+			SubxtClientFactory<CustomConfig>,
+			CustomConfig,
+			Metadata,
+			SubxtMetadataProvider<CustomConfig>,
+		>,
+	>,
+) -> Result<(), ()> {
+	let mut quote = vec![];
+	let mut attestation_type =
+		litentry_rococo::teebag::calls::types::register_enclave::AttestationType::Dcap(
+			teebag::types::DcapProvider::Intel,
+		);
+
+	#[cfg(feature = "gramine-quote")]
+	{
+		use std::fs;
+		use std::fs::File;
+		use std::io::Write;
+		let mut f = File::create("/dev/attestation/user_report_data").unwrap();
+		let content = signer.public_key().0;
+		f.write_all(&content).unwrap();
+
+		quote = fs::read("/dev/attestation/quote").unwrap();
```

Comment:
Do we know how to generate/mount this?

Reply:
its exposed by gramine
https://gramine.readthedocs.io/en/stable/attestation.html#low-level-dev-attestation-interface